matrix/common/etl/transformers/cell_expression.py

```diff
@@ -37,13 +41,18 @@ def _write_rows_to_psvs(self, *args: typing.Tuple):
 
     def _parse_from_metadatas(self, bundle_dir):
         if os.path.isfile(os.path.join(bundle_dir, "matrix.mtx")):
-            cell_lines, expression_lines = self._parse_10x_bundle(bundle_dir)
+            cell_lines, expression_lines = self._parse_cellranger_bundle(bundle_dir)
+        elif os.path.isfile(os.path.join(bundle_dir, "empty_drops_result.csv")):
```

Comment:
This feels like a risky way to get the analysis type now that we're starting to get more of them. I think a better way would be like `json.load(open("analysis_protocol_0.json"))["computational_method"] == "Optimus"`, but I'm checking with the pipelines team.

Reply:
Sounds good. Will update when we have an answer

Reply:
Okay, so based on their response I would do this:
```python
protocol_id = json.load(open(os.path.join(bundle_dir, "analysis_protocol_0.json")))["protocol_core"]["protocol_id"]
if protocol_id.startswith("optimus"):
    ....
elif protocol_id.startswith("cellranger"):
    ....
elif protocol_id.startswith("smartseq2"):
    ....
````

Reply:
Done